src/compositions/usePagination.ts

```diff
@@ -123,29 +127,27 @@ export function usePagination<
     page.value--
   }
 
-  function setFetchSubscriptionParameters(): void {
-    fetchSubscriptionParameters.value = getFetchSubscriptionParameters()
-  }
-
-  function getFetchSubscriptionParameters(): Parameters<TFetch>[] | null {
-    const parameters = fetchParametersGetter()
-
-    if (parameters === null) {
-      return null
+  function getPagesToFetch(page: number): number[] {
+    if (mode === 'page') {
+      return [page]
     }
 
-    const [filter, ...rest] = parameters
-    const filters = getFetchParametersForPages(page.value, filter)
-
-    return filters.map(filter => [filter, ...rest]) as unknown as Parameters<TFetch>[]
+    return repeat(page, index => index + 1)
   }
 
-  function getFetchParametersForPages(page: number, filter?: TFetchFilter): TFetchFilter[] {
-    if (mode === 'page') {
-      return [getFetchFilterForPage(page, filter)]
-    }
+  function getFetchParametersForPage(page: number): Ref<Parameters<TFetch> | null> {
+    return toRef(() => {
+      const parameters = fetchParametersGetter()
 
-    return repeat(page, index => getFetchFilterForPage(index + 1, filter))
+      if (parameters === null) {
+        return null
+      }
+
+      const [filter, ...rest] = parameters
+      const pageFilter = getFetchFilterForPage(page, filter)
+
+      return [pageFilter, ...rest]
+    }) as Ref<Parameters<TFetch> | null>
```

Comment:
Rather than generating every pages parameters statically via watchers this gets the parameters for a single page in a ref wrapped getter. This means every page has the ability to re-evaluate the parameters when the subscription is refreshed. Again, important for how relative time works.